compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -117,6 +119,47 @@ object ScenarioService {
     Status.NOT_FOUND.withDescription(s" context $id not found!").asRuntimeException
 }
 
+sealed trait ScriptStream {
+  def sendStatus(status: ScenarioStatus): Unit;
+  def sendFinalResponse(result: Either[ScenarioError, ScenarioResult]): Unit;
+  def sendError(t: Throwable): Unit;
+}
+
+final case class StreamWithoutStatus(internal: StreamObserver[RunScenarioResponse])
+    extends ScriptStream {
+  override def sendFinalResponse(finalResponse: Either[ScenarioError, ScenarioResult]): Unit = {
+    val message = finalResponse match {
+      case Left(error: ScenarioError) => RunScenarioResponse.newBuilder.setError(error).build
+      case Right(result: ScenarioResult) => RunScenarioResponse.newBuilder.setResult(result).build
+    }
+    internal.onNext(message)
+    internal.onCompleted()
+  }
+
+  override def sendStatus(status: ScenarioStatus): Unit = {}
+  override def sendError(t: Throwable): Unit = internal.onError(t)
+}
+
+final case class StreamWithStatus(internal: StreamObserver[RunScenarioResponseOrStatus])
+    extends ScriptStream {
+  override def sendFinalResponse(finalResponse: Either[ScenarioError, ScenarioResult]): Unit = {
+    val message = finalResponse match {
+      case Left(error: ScenarioError) =>
+        RunScenarioResponseOrStatus.newBuilder.setError(error).build
+      case Right(result: ScenarioResult) =>
+        RunScenarioResponseOrStatus.newBuilder.setResult(result).build
+    }
+    internal.onNext(message)
+    internal.onCompleted()
+  }
+
+  override def sendStatus(status: ScenarioStatus): Unit = {
+    val message = RunScenarioResponseOrStatus.newBuilder.setStatus(status).build
+    internal.onNext(message)
+  }
+  override def sendError(t: Throwable): Unit = internal.onError(t)
+}
```

Comment:
I personally prefer to but the `case class` in the companion object of type they extend.
Things are arranged more nicely and you do not need to prefix all your case class.
For instance if you define you classed like:
```suggestion
 object ScriptStream {
    final case class WithoutStatus(internal: StreamObserver[RunScenarioResponse])
      extends ScriptStream {
      override def sendFinalResponse(finalResponse: Either[ScenarioError, ScenarioResult]): Unit = {
        val message = finalResponse match {
          case Left(error: ScenarioError) => RunScenarioResponse.newBuilder.setError(error).build
          case Right(result: ScenarioResult) => RunScenarioResponse.newBuilder.setResult(result).build
        }
        internal.onNext(message)
        internal.onCompleted()
      }

      override def sendStatus(status: ScenarioStatus): Unit = {}

      override def sendError(t: Throwable): Unit = internal.onError(t)
    }

    final case class WithStatus(internal: StreamObserver[RunScenarioResponseOrStatus])
      extends ScriptStream {
      override def sendFinalResponse(finalResponse: Either[ScenarioError, ScenarioResult]): Unit = {
        val message = finalResponse match {
          case Left(error: ScenarioError) =>
            RunScenarioResponseOrStatus.newBuilder.setError(error).build
          case Right(result: ScenarioResult) =>
            RunScenarioResponseOrStatus.newBuilder.setResult(result).build
        }
        internal.onNext(message)
        internal.onCompleted()
      }

      override def sendStatus(status: ScenarioStatus): Unit = {
        val message = RunScenarioResponseOrStatus.newBuilder.setStatus(status).build
        internal.onNext(message)
      }

      override def sendError(t: Throwable): Unit = internal.onError(t)
    }
  }
```

You can use those like `ScriptStream.WithoutStatus`.

Reply:
Feel free to chose what you prefer.

Reply:
Oh I didn't notice that, cheers! Changed made :)